src/plugins/fileview/fileview.tsx

```diff
@@ -0,0 +1,121 @@
+
+class FileViewRendererModel {
+    context: T.RendererContext;
+    opts: T.RendererOpts;
+    api: T.RendererModelContainerApi;
+    savedHeight: number;
+    updateHeight_debounced: (newHeight: number) => void;
+    ptyDataSource: (termContext: T.TermContextUnion) => Promise<T.PtyDataType>;
+    rawCmd: T.WebCmd;
+    loading: OV<boolean>;
+    isDone: OV<boolean>;
+    dirList: mobx.IObservableArray<any>;
+
+    constructor() {
+        this.updateHeight_debounced = debounce(1000, this.updateHeight.bind(this));
+    }
+
+    initialize(params: T.RendererModelInitializeParams): void {
+        this.loading = mobx.observable.box(true, { name: "renderer-loading" });
+        this.isDone = mobx.observable.box(params.isDone, { name: "renderer-isDone" });
+        this.context = params.context;
+        this.opts = params.opts;
+        this.api = params.api;
+        this.savedHeight = params.savedHeight;
+        this.ptyDataSource = params.ptyDataSource;
+        this.rawCmd = params.rawCmd;
+        this.dirList = mobx.observable.array(null, {
+            name: "FileView-directorylist",
+        });
+    }
+
+    dispose(): void {
+        return;
+    }
+
+    giveFocus(): void {
+        return;
+    }
+
+    updateOpts(update: T.RendererOptsUpdate): void {
+        Object.assign(this.opts, update);
+    }
+
+    setIsDone(): void {
+        if (this.isDone.get()) {
+            return;
+        }
+        mobx.action(() => {
+            this.isDone.set(true);
+        })();
+    }
+
+    reload(delayMs: number): void {
+        return;
+    }
+
+    updateHeight(newHeight: number): void {
+        if (this.savedHeight != newHeight) {
+            this.savedHeight = newHeight;
+            this.api.saveHeight(newHeight);
+        }
+    }
+
+    parseBytesToUTF(data: Uint8Array): string {
+        let rtn = "";
+        for (let index = 0; index < data.length; index++) {
+            let curByte = data[index];
+            rtn += String.fromCharCode(curByte);
+        }
+        return rtn;
+    }
+
+    parseUTFPacketStringJSON(packetString: string) {
+        let jsonStartIndex = packetString.indexOf("{");
+        let jsonString = packetString.substring(jsonStartIndex, packetString.length).trim();
+        let jsonPk = JSON.parse(jsonString);
+        return jsonPk;
+    }
+
+    receiveData(pos: number, data: Uint8Array, reason?: string): void {
+        let packetString = this.parseBytesToUTF(data);
+        let pk = this.parseUTFPacketStringJSON(packetString);
+        console.log("pk:", pk.name);
+        mobx.action(() => {
+            this.dirList.push(pk);
+        })();
+    }
+}
+
+class FileViewRenderer extends React.Component<{ model: FileViewRendererModel }> {
+    renderFile(file: any, index: number) {
+        let keyString = "file-" + index;
+        return <div key={keyString}>{file.name}</div>;
+    }
+
+    render() {
+        let model: FileViewRendererModel = this.props.model;
+        let dirList = model.dirList;
```

Comment:
how do I trigger the re render here?
Now I can see in the logs that all the dirs get sent almost immediately, but I have to trigger a redraw otherwise, for example by clicking on the line

Reply:
Resolved